jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -0,0 +1,132 @@
+import base64
+import os
+import re
+import uuid
+
+from astropy.io import fits
+from astropy.wcs import WCS
+from glue.core.data import Component, Data
+try:
+    from jwst import datamodels
+    HAS_JWST_ASDF = True
+except ImportError:
+    HAS_JWST_ASDF = False
+
+from jdaviz.core.registries import data_parser_registry
+
+__all__ = ['parse_data']
+
+
+@data_parser_registry("imviz-data-parser")
+def parse_data(app, file_obj, data_label=None, show_in_viewer=True):
+    """Parse a data file into Imviz.
+
+    Parameters
+    ----------
+    app : `~jdaviz.app.Application`
+        The application-level object used to reference the viewers.
+
+    file_path : str
+        The path to an image data file.
+
+    data_label : str, optional
+        The label to be applied to the Glue data component.
+
+    show_in_viewer : bool
+        Show data in viewer.
+
+    """
+    # TODO: How much support needed for URL/URI?
```

Comment:
Doesn't astropy.io.fits handle URLs transparently?

Reply:
Yes, but as discussed offline, Glue/Jdaviz cannot handle arbitrary filenames (e.g., output from `astropy.utils.data.download_file`). It is a battle for another day, so I'll leave this comment in for now.

Reply:
OK, I made some changes to the parser. I think as long as `fits` or `jwst` can handle URL/URI, it should be no problem.